airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -172,27 +177,89 @@ def _doc_upload(
     return doc_uploaded, doc_blob_id
 
 
-def create_prerelease_metadata_file(metadata_file_path: Path, validator_opts: ValidatorOptions) -> Path:
-    metadata, error = validate_and_load(metadata_file_path, [], validator_opts)
-    if metadata is None:
-        raise ValueError(f"Metadata file {metadata_file_path} is invalid for uploading: {error}")
+def _apply_prerelease_overrides(metadata_dict: dict, validator_opts: ValidatorOptions) -> dict:
+    """Apply any prerelease overrides to the metadata file before uploading it to GCS."""
+    if validator_opts.prerelease_tag is None:
+        return metadata_dict
 
     # replace any dockerImageTag references with the actual tag
     # this includes metadata.data.dockerImageTag, metadata.data.registries[].dockerImageTag
     # where registries is a dictionary of registry name to registry object
-    metadata_dict = to_json_sanitized_dict(metadata, exclude_none=True)
     metadata_dict["data"]["dockerImageTag"] = validator_opts.prerelease_tag
     for registry in get(metadata_dict, "data.registries", {}).values():
         if "dockerImageTag" in registry:
             registry["dockerImageTag"] = validator_opts.prerelease_tag
 
-    # write metadata to yaml file in system tmp folder
-    tmp_metadata_file_path = Path("/tmp") / metadata.data.dockerRepository / validator_opts.prerelease_tag / METADATA_FILE_NAME
-    tmp_metadata_file_path.parent.mkdir(parents=True, exist_ok=True)
-    with open(tmp_metadata_file_path, "w") as f:
-        yaml.dump(metadata_dict, f)
+    return metadata_dict
+
+
+def _get_git_info_for_file(original_metadata_file_path: Path) -> Optional[GitInfo]:
+    """
+    Add additional information to the metadata file before uploading it to GCS.
+
+    e.g. The git commit hash, the date of the commit, the author of the commit, etc.
+
+    """
+    repo = git.Repo(search_parent_directories=True)
```

Comment:
I'd appreciate if the metadata service could avoid direct `git` interactions as it will make it required to run on top of a checked out repo. I would prefer to defer the "git log" command execution to the client and make the metadata service only parse a git log output.

This suggestion comes from the kind of difficulties I spotted while trying to make `airbyte-ci` target any repo.

Reply:
Fair call out! I can change to support that approach

Reply:
Sounds like a reasonable tradeoff 👍 . Can we log a `warning` when no repo was found? (Would it mean that you'd move your `try` up by one line?

Reply:
Absolutely